src/Hl7.Fhir.Base/FhirPath/Expressions/Closure.cs

```diff
@@ -82,5 +98,13 @@ public virtual IEnumerable<ITypedElement> ResolveValue(string name)
 
             return null;
         }
+
+        private static ITypedElement getResourceFromNode(ScopedNode node) => node.AtResource ? node : node.ParentResource;
```

Comment:
I think we should return ScopedNode here, since in this function we have first hand knowledge that if you pass in a ScopedNode, node and node.ParentResource are ScopedNodes too. This way you can avoid the cast when you call this function, which looks far more suspicious.